core/types/transaction.go

```diff
@@ -227,7 +229,11 @@ func (tx *Transaction) ProtoEncode() (*ProtoTransaction, error) {
 			workNonce := tx.WorkNonce().Uint64()
 			protoTx.WorkNonce = &workNonce
 		}
-
+		if tx.Data() == nil {
```

Comment:
I think we may need to do this in Quais.js else the sig may not match.

Reply:
That is correct